session/handler.go

```diff
@@ -215,13 +215,20 @@ func (h *Handler) whoami(w http.ResponseWriter, r *http.Request, _ httprouter.Pa
 	s, err := h.r.SessionManager().FetchFromRequest(ctx, r)
 	c := h.r.Config()
 	if err != nil {
+		h.r.Audit().WithRequest(r).WithError(err).Info("No valid session found.")
+
 		// We cache errors (and set cache header only when configured) where no session was found.
-		if noSess := new(ErrNoActiveSessionFound); c.SessionWhoAmICaching(ctx) && errors.As(err, &noSess) && noSess.credentialsMissing {
-			w.Header().Set("Ory-Session-Cache-For", fmt.Sprintf("%d", int64(time.Minute.Seconds())))
+		if noSess := new(ErrNoActiveSessionFound); errors.As(err, &noSess) {
+			if c.SessionWhoAmICaching(ctx) && noSess.credentialsMissing {
+				w.Header().Set("Ory-Session-Cache-For", fmt.Sprintf("%d", int64(time.Minute.Seconds())))
+			}
+
+			h.r.Writer().WriteError(w, r, ErrNoSessionFound.WithWrap(err))
+			return
+
 		}
 
-		h.r.Audit().WithRequest(r).WithError(err).Info("No valid session found.")
-		h.r.Writer().WriteError(w, r, ErrNoSessionFound.WithWrap(err))
+		h.r.Writer().WriteError(w, r, herodot.ErrInternalServerError.WithReasonf("Unable to validate session.").WithWrap(err))
```

Comment:
Kratos was doing this before so probably not a huge problem, but I guess there's no concern of exposing the error in the response? It seems like internals could be revealed by spitting out the error as is.

Reply:
It actually does not spit out the internals other than in the logs. It just says "Unable to validate session.".

Many other endpoints do actually expose the internals, but the `whoami` endpoint is the only one I wanted to address.

Reply:
Ah okay. Thanks for clarifying.